lib/Munin/Master/UpdateWorker.pm

```diff
@@ -499,8 +499,12 @@ sub _db_state_update {
 		WHERE ds.name = ?");
 	$sth_ds->execute($node_id, $plugin, $field);
 	my ($ds_id) = $sth_ds->fetchrow_array();
-	DEBUG "_db_state_update.ds_id:$ds_id";
 	$sth_ds->finish();
+	if (! defined $ds_id) {
+		$dbh->disconnect();
+		return undef;
+	}
+	DEBUG "_db_state_update.ds_id:$ds_id";
```

Comment:
if `$ds_id` is `undef` it is more because the data in the DB is wrong.

So we don't need to disconnect. Actually, we should never `$dbh->disconnect()` at random places in the code...